src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/HostBuilderContext.cs

```diff
@@ -11,24 +11,24 @@ namespace Microsoft.Extensions.Hosting
     /// </summary>
     public class HostBuilderContext
     {
-        public HostBuilderContext(IDictionary<object, object> properties!!)
+        public HostBuilderContext(IDictionary<object, object?> properties!!)
         {
             Properties = properties;
         }
 
         /// <summary>
         /// The <see cref="IHostEnvironment" /> initialized by the <see cref="IHost" />.
         /// </summary>
-        public IHostEnvironment HostingEnvironment { get; set; }
+        public IHostEnvironment? HostingEnvironment { get; set; }
```

Comment:
These two properties are always set (when we create the object):

https://github.com/dotnet/runtime/blob/385fc8832b1d4160abe13645054b65458b6a12f8/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L213-L217

It feels wrong to have consumers need to check these for null, when they won't be.

Reply:
Hm, why they are not part of constructor then?